snuba/datasets/metrics_messages.py

```diff
@@ -141,3 +172,23 @@ def aggregation_options_for_counter_message(
         apply_aggregation_option(settings, parsed_aggregation_setting)
 
     return settings
+
+
+def aggregation_options_for_gauge_message(
+    message: Mapping[str, Any], retention_days: int
+) -> Mapping[str, Any]:
+    settings = {
+        "granularities": [
+            GRANULARITY_ONE_MINUTE,
+            GRANULARITY_ONE_HOUR,
+            GRANULARITY_ONE_DAY,
+        ],
+        "min_retention_days": retention_days,
+        "materialization_version": 2,
+    }
+
+    if aggregation_setting := message.get("aggregation_option"):
+        parsed_aggregation_setting = AggregationOption(aggregation_setting)
+        apply_aggregation_option(settings, parsed_aggregation_setting)
```

Comment:
What if upstream provides an aggregation option of HIST? We should make this more robust such that certain aggregation options only apply to certain datasets.